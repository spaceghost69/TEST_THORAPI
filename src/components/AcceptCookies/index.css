.cookie-policy-modal {
  width: 500px;
}

.cookie-policy-modal .modal-body {
  padding: 20px;
}