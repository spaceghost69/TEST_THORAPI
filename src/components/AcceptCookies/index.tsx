import React, { useEffect, useState } from "react";
import { Accordion, Col, Row, Toast, ToastContainer } from "react-bootstrap";
import { useCookies } from "react-cookie";
import { FaHandshake, FaHandSpock } from "react-icons/fa";
import CoolButton from "../CoolButton";

interface AcceptCookiesProps {
  show: boolean;
  onHide: () => void;
}

const AcceptCookies: React.FC<AcceptCookiesProps> = () => {
  const [accepted, setAccepted] = useState(false);
  const [refused, setRefused] = useState(false);
  const [show, setShow] = useState(true);
  const [cookie, setCookie] = useCookies(["accept-cookie-policy"]);

  let acceptedCookie = (cookie["accept-cookie-policy"] === "accepted");

  useEffect(() => {
    if (acceptedCookie) {
      setAccepted(true);
    }
    if (show && acceptedCookie) {
      setTimeout(() => { setShow(false) }, 3000);
    }
    if (!show) {
      setShow(false);
    }
    if (cookie["accept-cookie-policy"] === "NOT_ACCEPTED") {
      setRefused(true);
    }
  }, [cookie, show]);

  return (
    <ToastContainer
      className="p-3"
      position="bottom-end"
      style={{ zIndex: 1 }}
    >
      <Toast onClose={() => setShow(false)} show={show} delay={3000} bg="dark">

        <Toast.Header>
          {!accepted && (
            <>
              {!refused && (
                <FaHandshake size={48} />
              )}
              {refused && !accepted && (
                <FaHandSpock size={48} />
              )}
              <strong className="me-auto"><div style={{ marginLeft: "1em", width: '100%', fontWeight: 'bold' }}>Please Read and Accept our TOS</div></strong>
            </>
          )}
        </Toast.Header>

        <Toast.Body>

          {accepted && (
            <h5>Welcome!</h5>
          )}


          {!accepted && refused && (

            <h3>You will need to accept the TOS to use our services</h3>

          )}


          {!accepted && (

            <Accordion flush>
              <Accordion.Item
                eventKey="privacy"
              >
                <Accordion.Header><b>Privacy Policy</b></Accordion.Header>
                <Accordion.Body>
                  We value your privacy and your data. We do not sell your data to third parties.
                  <br /><br />
                  For more details on how we use your data, please review our

                  <a
                    href="https://valkyrlabs.com/v1/docs/Legal/privacy/"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <br />
                    <b>Privacy Policy</b>
                  </a>
                </Accordion.Body>
              </Accordion.Item>
              <Accordion.Item
                eventKey="cookies"
              >


                <Accordion.Header><b>Terms of Service</b></Accordion.Header>
                <Accordion.Body>
                  We and our partners use cookies for essential site functionality, analytics, and to improve your user experience.

                  <br />
                  By using this site, you agree to our{" "}
                  <br />
                  <a
                    href="https://valkyrlabs.com/v1/docs/Legal/tos"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <b>Terms of Service</b>
                  </a>
                </Accordion.Body>
              </Accordion.Item>

            </Accordion>

          )}

          <br />

          <Row>
            <Col>
              {!accepted && (
                <CoolButton
                  variant={"info"}
                  onClick={() => {
                    setCookie("accept-cookie-policy", "accepted", { path: "/", maxAge: (1000 * 60 * 60 * 24 * 365 * 10) });
                    setAccepted(true);
                    setTimeout(() => setShow(false), 5000);
                  }}
                >
                  ACCEPT
                </CoolButton>
              )}
            </Col>
            <Col>
              {!accepted && (
                <CoolButton
                  variant={"warning"}
                  onClick={() => {
                    setCookie("accept-cookie-policy", "NOT_ACCEPTED", { path: "/" });
                    setAccepted(false);
                    setRefused(true);
                  }}
                >
                  DECLINE
                </CoolButton>
              )}
            </Col>
          </Row>




        </Toast.Body>
      </Toast>
    </ToastContainer >
  );
};

export default AcceptCookies;
