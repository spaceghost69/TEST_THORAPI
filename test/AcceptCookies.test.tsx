import { describe, expect, it } from '@jest/globals';
import '@testing-library/jest-dom';
import React from "react";
import renderer from "react-test-renderer";
import AcceptCookies from "../src/components/AcceptCookies";


describe("Snapshot testing of AcceptCookies", () => {
  it("AcceptCookies snapshot", () => {
    const AcceptCookiesT = renderer
      .create(
        <AcceptCookies
          show={true}
          onHide={() => { }}
        />
      )
      .toJSON();
    expect(AcceptCookiesT).toMatchSnapshot();
  });
});