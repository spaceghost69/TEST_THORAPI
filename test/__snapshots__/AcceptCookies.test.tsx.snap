// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing of AcceptCookies AcceptCookies snapshot 1`] = `
<div
  className="toast-container bottom-0 end-0 p-3"
  style={
    {
      "zIndex": 1,
    }
  }
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast bg-dark show"
    role="alert"
  >
    <div
      className="toast-header"
    >
      <svg
        fill="currentColor"
        height={48}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 640 512"
        width={48}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M434.7 64h-85.9c-8 0-15.7 3-21.6 8.4l-98.3 90c-.1.1-.2.3-.3.4-16.6 15.6-16.3 40.5-2.1 56 12.7 13.9 39.4 17.6 56.1 2.7.1-.1.3-.1.4-.2l79.9-73.2c6.5-5.9 16.7-5.5 22.6 1 6 6.5 5.5 16.6-1 22.6l-26.1 23.9L504 313.8c2.9 2.4 5.5 5 7.9 7.7V128l-54.6-54.6c-5.9-6-14.1-9.4-22.6-9.4zM544 128.2v223.9c0 17.7 14.3 32 32 32h64V128.2h-96zm48 223.9c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zM0 384h64c17.7 0 32-14.3 32-32V128.2H0V384zm48-63.9c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16c0-8.9 7.2-16 16-16zm435.9 18.6L334.6 217.5l-30 27.5c-29.7 27.1-75.2 24.5-101.7-4.4-26.9-29.4-24.8-74.9 4.4-101.7L289.1 64h-83.8c-8.5 0-16.6 3.4-22.6 9.4L128 128v223.9h18.3l90.5 81.9c27.4 22.3 67.7 18.1 90-9.3l.2-.2 17.9 15.5c15.9 13 39.4 10.5 52.3-5.4l31.4-38.6 5.4 4.4c13.7 11.1 33.9 9.1 45-4.7l9.5-11.7c11.2-13.8 9.1-33.9-4.6-45.1z"
        />
      </svg>
      <strong
        className="me-auto"
      >
        <div
          style={
            {
              "fontWeight": "bold",
              "marginLeft": "1em",
              "width": "100%",
            }
          }
        >
          Please Read and Accept our TOS
        </div>
      </strong>
      <button
        aria-label="Close"
        className="btn-close"
        data-dismiss="toast"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="toast-body"
    >
      <div
        className="accordion accordion-flush"
      >
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              <b>
                Privacy Policy
              </b>
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              We value your privacy and your data. We do not sell your data to third parties.
              <br />
              <br />
              For more details on how we use your data, please review our
              <a
                href="https://valkyrlabs.com/v1/docs/Legal/privacy/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <br />
                <b>
                  Privacy Policy
                </b>
              </a>
            </div>
          </div>
        </div>
        <div
          className="accordion-item"
        >
          <h2
            className="accordion-header"
          >
            <button
              aria-expanded={false}
              className="accordion-button collapsed"
              onClick={[Function]}
              type="button"
            >
              <b>
                Terms of Service
              </b>
            </button>
          </h2>
          <div
            aria-expanded={null}
            className="accordion-collapse collapse"
          >
            <div
              className="accordion-body"
            >
              We and our partners use cookies for essential site functionality, analytics, and to improve your user experience.
              <br />
              By using this site, you agree to our
               
              <br />
              <a
                href="https://valkyrlabs.com/v1/docs/Legal/tos"
                rel="noopener noreferrer"
                target="_blank"
              >
                <b>
                  Terms of Service
                </b>
              </a>
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col"
        >
          <button
            className="btnx btn btn-info"
            onClick={[Function]}
          >
            ACCEPT
          </button>
        </div>
        <div
          className="col"
        >
          <button
            className="btnx btn btn-warning"
            onClick={[Function]}
          >
            DECLINE
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
